### PR TITLE
Update PangolinV2 contract

### PR DESCRIPTION
**Pull request by AABeely**
As per the discussion on discord, I've ammended the PangolinV2 contract with the following changes:

- Rename the harestWithCallFeeRecipient function to the latest convention.
- Add support for a optional second reward token route which can be set either at the time of deployment or at a later date without requiring the strategy contracts to be completely redeployed.

### CODE REVIEW

File: contracts/BIFI/strategies/Pangolin/StrategyPangolinMiniChefLP.sol [hunk 1]
```diff
@@ -266,13 +293,37 @@ contract StrategyPangolinMiniChefLP is StratManager, FeeManager {
...
+    }
+
+    function setRewardRoute(address[] memory _rewardToOutputRoute) external onlyOwner {
+        require(_rewardToOutputRoute[_rewardToOutputRoute.length - 1] == output, "rewardToOutputRoute[last] != output");
+        rewardToOutputRoute = _rewardToOutputRoute;
+        IERC20(_rewardToOutputRoute[0]).safeApprove(unirouter, 0);
+        IERC20(_rewardToOutputRoute[0]).safeApprove(unirouter, uint256(-1));
+    }
+
+    function removeRewardRoute() external onlyOwner {
```
Comment (sircryptomaxi) [inline contracts/BIFI/strategies/Pangolin/StrategyPangolinMiniChefLP.sol:321]: This function can be `onlyManager` instead so it doesn't need to go through the multi-sig.

File: contracts/BIFI/strategies/Pangolin/StrategyPangolinMiniChefLP.sol [hunk 2]
```diff
@@ -134,7 +146,16 @@ contract StrategyPangolinMiniChefLP is StratManager, FeeManager {
     // compounds earnings and charges performance fee
     function _harvest(address callFeeRecipient) internal whenNotPaused {
         IPangolinMiniChef(chef).harvest(poolId, address(this));
+        if (rewardToOutputRoute[0] != nullAddress) {
+                address reward = rewardToOutputRoute[0];
+                uint256 rewardBal = IERC20(reward).balanceOf(address(this));
+                if (rewardBal > 0) {
+                    IUniswapRouterETH(unirouter).swapExactTokensForTokens(rewardBal, 0, rewardToOutputRoute, address(this), now);
+                }
```
Comment (sircryptomaxi) [inline contracts/BIFI/strategies/Pangolin/StrategyPangolinMiniChefLP.sol:154]: Oddly tabbed

File: contracts/BIFI/strategies/Pangolin/StrategyPangolinMiniChefLP.sol [hunk 3]
```diff
@@ -266,13 +293,37 @@ contract StrategyPangolinMiniChefLP is StratManager, FeeManager {
 
         IERC20(lpToken1).safeApprove(unirouter, 0);
         IERC20(lpToken1).safeApprove(unirouter, type(uint256).max);
+
+        if (rewardToOutputRoute[0] != nullAddress) {
+                IERC20(rewardToOutputRoute[0]).safeApprove(unirouter, 0);
+                IERC20(rewardToOutputRoute[0]).safeApprove(unirouter, uint256(-1));
```
Comment (sircryptomaxi) [inline contracts/BIFI/strategies/Pangolin/StrategyPangolinMiniChefLP.sol:299]: Double tab again, not needed

**Comment by AABeely** [comment]
Pushed up changes to address the comments above

**Pull request merged**